src/library/browse/browsefeature.cpp

```diff
@@ -252,6 +257,11 @@ void BrowseFeature::activateChild(const QModelIndex& index) {
 }
 
 void BrowseFeature::onRightClickChild(const QPoint& globalPos, QModelIndex index) {
+    // Create the right-click menu and parent it to the sidebar widget in
+    // order to make it stylable with skin stylesheet rather than ugly OS styling.
+    // Parent to default NULL if there's no sidebar widget.
+    m_pMenu = new QMenu(m_pSidebarWidget);
```

Comment:
It looks like as if you create a new QMenu on every right click.
Doesn't this work?
QMenu menu(m_pSidebarWidget);

Reply:
nope, `QMenu menu(m_pSidebarWidget);` doesn't work. idk why

Is there a problem with creating a new menu? the old one is overwritten so there shouldn't be any side effect like increased memory usage, right?

Reply:
Only the plain pointer to the memory region is overwritten. The memory itself remains.  
You should use a smart pointer e.g. unique_ptr which automatically deletes the managed object if it is replaced see: https://de.cppreference.com/w/cpp/memory/unique_ptr/reset

Reply:
Instead of using a unique pointer (which I don't really know how to handle, sry) I could also move `m_pMenu = new QMenu(m_pSidebarWidget);` to `void SetlogFeature::bindSidebarWidget` and clear/pouplate the menu in `void SetlogFeature::onRightClickChild`, right?

Reply:
Sounds good. "new" is then called only once, right?

Reply:
jup, once.
when `m_pSetlogFeature->bindSidebarWidget(pSidebarWidget);` is called from library.cpp when the feature is added.